common/table.js

```diff
@@ -426,24 +411,336 @@
+                    read(scope);
+                }
+
+            };
+
+            scope.toggleSortOrder = function () {
+                scope.vm.sortOrder = (scope.vm.sortOrder === 'asc' ? scope.vm.sortOrder = 'desc' : scope.vm.sortOrder = 'asc');
+                scope.vm.reference = scope.vm.reference.sort([{"column":scope.vm.sortby, "descending":(scope.vm.sortOrder === "desc")}]);
+                read(scope);
+            };
+
+            scope.isDisabled = function (tuple) {
+                if (!scope.vm.disabledRows || scope.vm.disabledRows.length == 0) {
+                    return false;
+                }
+
+                return scope.vm.disabledRows.findIndex(function (obj) {
+                    return obj.uniqueId == tuple.uniqueId;
+                }) > -1;
+            };
+
+            // verifies whether or not the current key value is in the set of selected rows or not
+            scope.isSelected = function (tuple) {
+                var index = scope.vm.selectedRows.findIndex(function (obj) {
+                    return obj.uniqueId == tuple.uniqueId;
+                });
+                return (index > -1);
+            };
+
+            // this is for the button on the table heading that deselects all currently visible rows
+            scope.selectNone = function() {
+                var tuples = [], tuple;
+                for (var i = 0; i < scope.vm.page.tuples.length; i++) {
+                    tuple = scope.vm.page.tuples[i];
+                    var key = tuple.uniqueId;
+
+                    var index = scope.vm.selectedRows.findIndex(function (obj) {
+                        return obj.uniqueId == key;
+                    });
+
+                    if (index > -1) {
+                        tuples.push(tuple);
+                        scope.vm.selectedRows.splice(index, 1);
+                    }
+                }
+                if (tuples.length > 0) {
+                    callOnRowClick(scope, tuples, false);
+                }
+            };
+
+            // this is for the button on the table heading that selects all currently visible rows
+            scope.selectAll = function() {
+                var tuples = [], tuple;
+                for (var i = 0; i < scope.vm.page.tuples.length; i++) {
+                    var tuple = scope.vm.page.tuples[i];
+
+                    if (scope.isDisabled(tuple)) continue;
+
+                    if (!scope.isSelected(tuple)) {
+                        scope.vm.selectedRows.push(tuple);
+                        tuples.push(tuple);
+                    }
+                }
+                if (tuples.length > 0) {
+                    callOnRowClick(scope, tuples, true);
+                }
+            };
+
+
+            // Facilitates the multi select functionality for multi edit by storing the tuple in the selectedRows array
+            scope.onSelect = function(args) {
+                console.log(args);
```

Comment:
remove please. Or you can change it to $log with debug flag so it only shows up in that mode.

Reply:
I forgot to remove this. I added this to test that this way of extending the directive works or not.